packages/platform-core/src/usage-data/usage_data_emitter_factory.ts

```diff
@@ -19,6 +21,15 @@ export class UsageDataEmitterFactory {
    * Returns a ClientConfigGenerator for the given BackendIdentifier type
    */
   getInstance = (libraryVersion: string): UsageDataEmitter => {
+    const configController = new ConfigController();
+
+    if (
+      process.env.AMPLIFY_DISABLE_TELEMETRY ||
+      configController.get<boolean>(TELEMETRY_ENABLED_KEY) === false
```

Comment:
Do we need environment variable?

Reply:
The env var is primary way that the user can disable telemetry in CI environments.